### ISSUE TITLE
Difference between eqx.static_field = True and jax.lax.stop_gradient

### ISSUE THREAD

**Issue by timnotavailable** [patrick-kidger/equinox#957]
Hello, 
 I'm going to make a ODE simulator for Bloch Equation, I would like to define a module which has some attributes of array type, 

class(eqx.Module):
  array1:Array["x y z"]
  array2: Array ["x y z"]

what I would like to do is : When I'm initializing with a np.array, then it is static and do not need the gradient, when initializing with a jnp.array then it is marked dynamic in jax (requires gradient). Currently I looked the whole document, and there are two ways doing so : eqx.static_field = True or   jax.lax.stop_gradient , which one should I choose and how should I wrap this function into the initialization? 

I've also checked some issues, it is said that  eqx.static_field = True this way is dangerous, what should I do to realize it?
